### PR TITLE
fix(AwsVpcPeering): relaxes requeuing while waiting peering statuses

### PR DESCRIPTION
**Pull request by vandjelk**
**Description**

Changes proposed in this pull request:

- adding tests for VpcPeering deletion when unauthorized in remote
- relaxes requeuing while waiting for peering pending acceptance
- relaxes requeuing while waiting for peering active

### CODE REVIEW

File: internal/controller/cloud-control/vpcpeering_aws_test.go [hunk 1]
```diff
@@ -657,4 +650,231 @@ var _ = Describe("Feature: KCP VpcPeering", func() {
...
+		By("And Then there are no additional route entries added to remote route tables", func() {
+			Expect(awsMockRemote.GetRouteCount(infra.Ctx(), remoteVpcId, kcpPeering.Status.RemoteId, localVpcCidr)).
+				To(Equal(2))
+		})
+
+		By("And Then remote VpcPeeringConnection is not deleted", func() {
+			remotePeering, _ := awsMockRemote.DescribeVpcPeeringConnection(infra.Ctx(), kcpPeering.Status.Id)
+			Expect(remotePeering).NotTo(BeNil())
+		})
+
```
Comment (dushanpantic) [inline internal/controller/cloud-control/vpcpeering_aws_test.go:842]: In my opinion, this whole section is not necessary for this scenario
```suggestion
```

File: internal/controller/cloud-control/vpcpeering_aws_test.go [hunk 2]
```diff
@@ -657,4 +650,231 @@ var _ = Describe("Feature: KCP VpcPeering", func() {
...
+		By("When remote permissions are removed", func() {
```
Comment (dushanpantic) [inline internal/controller/cloud-control/vpcpeering_aws_test.go:843]:
```suggestion
		By("And Given remote permissions are removed", func() {
```

File: internal/controller/cloud-control/vpcpeering_aws_test.go [hunk 3]
```diff
@@ -657,4 +650,231 @@ var _ = Describe("Feature: KCP VpcPeering", func() {
 				To(Equal(2))
 		})
 	})
+
+	It("Scenario: KCP AWS VpcPeering can be deleted when unauthorized", func() {
```
Comment (dushanpantic) [inline internal/controller/cloud-control/vpcpeering_aws_test.go:654]:
```suggestion
	It("Scenario: KCP AWS VpcPeering can be deleted when remote VPC Network authorization is revoked", func() {
```

File: pkg/kcp/provider/aws/vpcpeering/createVpcPeeringConnection.go [hunk 4]
```diff
@@ -101,6 +101,9 @@ func createVpcPeeringConnection(ctx context.Context, st composed.State) (error,
 	state.vpcPeering = vpcPeering
 
 	state.ObjAsVpcPeering().Status.Id = ptr.Deref(vpcPeering.VpcPeeringConnectionId, "")
+	state.ObjAsVpcPeering().Status.VpcId = ptr.Deref(state.vpc.VpcId, "")
+
+	meta.RemoveStatusCondition(state.ObjAsVpcPeering().Conditions(), cloudcontrolv1beta1.ConditionTypeError)
```
Comment (dushanpantic) [inline pkg/kcp/provider/aws/vpcpeering/createVpcPeeringConnection.go:106]: why is this condition removed here?

File: internal/controller/cloud-control/vpcpeering_aws_test.go (hunk 1 above)
Comment (vandjelk) [reply internal/controller/cloud-control/vpcpeering_aws_test.go:842]: Correct! This is already covered in different scenario.

File: pkg/kcp/provider/aws/vpcpeering/createVpcPeeringConnection.go (hunk 4 above)
Comment (vandjelk) [reply pkg/kcp/provider/aws/vpcpeering/createVpcPeeringConnection.go]: moved to updateStatusSuccess

File: internal/controller/cloud-control/vpcpeering_aws_test.go [hunk 5]
```diff
@@ -657,4 +646,234 @@ var _ = Describe("Feature: KCP VpcPeering", func() {
...
+			localKcpNet = cloudcontrolv1beta1.NewNetworkBuilder().
+				WithScope(scope.Name).
+				WithAwsRef(scope.Spec.Scope.Aws.AccountId, scope.Spec.Region, scope.Spec.Scope.Aws.Network.VPC.Id, localKcpNetworkName).
+				Build()
+			Eventually(CreateObj).
+				WithArguments(infra.Ctx(), infra.KCP().Client(), localKcpNet, WithName(localKcpNetworkName)).
+				Should(Succeed())
+		})
+
+		By("And remote KCP Network is Ready", func() {
```
Comment (dushanpantic) [inline internal/controller/cloud-control/vpcpeering_aws_test.go:753]:
```suggestion
		By("And Given remote KCP Network is Ready", func() {
```

File: internal/controller/cloud-control/vpcpeering_aws_test.go [hunk 6]
```diff
@@ -657,4 +646,234 @@ var _ = Describe("Feature: KCP VpcPeering", func() {
...
+			remoteKcpNet = cloudcontrolv1beta1.NewNetworkBuilder().
+				WithScope(scope.Name).
+				WithAwsRef(remoteAccountId, remoteRegion, remoteVpcId, remoteVpcName).
+				Build()
+			Eventually(CreateObj).
+				WithArguments(infra.Ctx(), infra.KCP().Client(), remoteKcpNet, WithName(remoteKcpNetworkName)).
+				Should(Succeed())
+		})
+
+		By("And remote KCP Network is Ready", func() {
```
Comment (dushanpantic) [inline internal/controller/cloud-control/vpcpeering_aws_test.go:774]:
```suggestion
		By("And Given remote KCP Network is Ready", func() {
```

File: internal/controller/cloud-control/vpcpeering_aws_test.go [hunk 7]
```diff
@@ -657,4 +646,234 @@ var _ = Describe("Feature: KCP VpcPeering", func() {
...
+
+		By("And Given VpcPeering is Ready", func() {
+			Eventually(LoadAndCheck, "2s").
+				WithArguments(infra.Ctx(), infra.KCP().Client(), kcpPeering,
+					NewObjActions(),
+					HavingConditionTrue(cloudcontrolv1beta1.ConditionTypeReady)).
+				Should(Succeed())
+		})
+
+		By("And Then remote VpcPeeringConnection is not deleted", func() {
```
Comment (dushanpantic) [inline internal/controller/cloud-control/vpcpeering_aws_test.go:837]:
```suggestion
		By("And Given remote VpcPeeringConnection exists", func() {
```

**Review by dushanpantic — APPROVED** [review]

**Pull request merged**